doc/source/user_guide/tutorials/mesh/create_a_mesh_from_scratch.rst

```diff
@@ -0,0 +1,158 @@
+.. _ref_tutorials_create_a_mesh_from_scratch:
+
+==========================
+Create a mesh from scratch
+==========================
+
+.. include:: ../../../links_and_refs.rst
+
+This tutorial demonstrates how to build a |MeshedRegion| from the scratch.
+
+The mesh object in DPF is a |MeshedRegion|. You can create your own |MeshedRegion| object and use it
+with DPF operators. The ability to use scripting to create any DPF entity means
+that you are not dependent on result files and can connect the DPF environment
+with any Python tool.
+
+In this tutorial, we create a parallel piped mesh made of linear hexa elements.
+
+:jupyter-download-script:`Download tutorial as Python script<create_a_mesh_from_scratch>`
+:jupyter-download-notebook:`Download tutorial as Jupyter notebook<create_a_mesh_from_scratch>`
+
+Import the necessary modules
+----------------------------
+
+Import the ``ansys.dpf.core`` module, including the operators module and the numpy library.
+
+.. jupyter-execute::
+
+    # Import the numpy library
+    import numpy as np
+    # Import the ``ansys.dpf.core`` module
+    from ansys.dpf import core as dpf
+    # Import the operators module
+    from ansys.dpf.core import operators as ops
+
+Define the mesh dimensions
+--------------------------
+
+.. jupyter-execute::
+
+    # Define the mesh dimensions
+    length = 0.1
+    width = 0.05
+    depth = 0.1
+    num_nodes_in_length = 10
+    num_nodes_in_width = 5
+    num_nodes_in_depth = 10
+    # Create a MeshedRegion object
+    my_meshed_region = dpf.MeshedRegion()
+
+Define the connectivity function
+--------------------------------
+
+To create a mesh you must define the nodes connectivity. This means to define
+the nodes ids connected to each element.
+
+Here, we create a function that will find this connectivity.
+
+.. jupyter-execute::
+
+    def search_sequence_numpy(arr, seq):
+        """Find a sequence in an array and return its index."""
+        indexes = np.where(np.isclose(arr, seq[0]))
+        for index in np.nditer(indexes[0]):
+            if index % 3 == 0:
+                if np.allclose(arr[index + 1], seq[1]) and np.allclose(arr[index + 2], seq[2]):
+                    return index
+        return -1
```

Comment:
I think this should work too. It avoids using the `np.nditer` function and handles dimensions before executing any logic:

```suggestion
    if len(seq) == 0:
        raise ValueError("The search sequence must not be empty.")
    if len(seq) > len(arr):
        return -1  # Sequence longer than the array cannot be found.

    # Sliding window approach
    for i in range(len(arr) - len(seq) + 1):
        if np.allclose(arr[i:i + len(seq)], seq):
            return i
    return -1
```